# Get Started
Steps for a basic CyteType job to get started using a sample pbmc dataset from 10X Genomics as an example.

## Quick Start Example
``` R
# Load package
library(CyteTypeR)

prepped_data <- PrepareCyteTypeR(pbmc,
         pbmc.markers,
         n_top_genes = 10,
         group_key = 'seurat_clusters',
         aggregate_metadata = TRUE,
         coordinates_key = "umap")

metadata <- list(
  title = 'My scRNA-seq analysis of human pbmc',
  run_label = 'initial_analysis',
  experiment_name = 'pbmc_human_samples_study')

results <- CyteTypeR(obj=pbmc,
                     prepped_data = prepped_data, 
                     study_context = "pbmc blood samples from humans", 
                     metadata = metadata
                          )
```

## Pre-processing
The current version of CyteTypeR works with analyzed datasets in Seurat objects and marker tables generated by the ```FindAllMarkers()``` function. You'll need to complete some minimal pre-processing steps before running CyteTypeR.

An example of the expected marker genes table can be found here: [marker-table-example.tsv](/inst/marker-table-example.tsv)

``` R

# Load libraries
library(dplyr)
library(patchwork)
library(Matrix)
library(Seurat)
library(CyteTypeR)


# Load an example dataset from MTX format files from 10X
pbmc.data <- Read10X(data.dir = "./data/filtered_gene_bc_matrices/hg19/")

# Initialize the Seurat object with the raw (non-normalized data).
pbmc <- CreateSeuratObject(counts = pbmc.data, project = "pbmc3k", min.cells = 3, min.features = 200)
pbmc <- NormalizeData(pbmc, normalization.method = "LogNormalize", scale.factor = 10000)
pbmc <- FindVariableFeatures(pbmc, selection.method = "vst", nfeatures = 2000)

all.genes <- rownames(pbmc)
pbmc <- ScaleData(pbmc, features = all.genes)

## Cluster the cells and run UMAP  (IMPORTANT: this step is currently required for using CyteTypeR)
pbmc <- FindNeighbors(pbmc, dims = 1:10)
pbmc <- FindClusters(pbmc, resolution = 0.5)

pbmc <- RunUMAP(pbmc, dims = 1:10)

# Find markers for all Clusters (IMPORTANT: this step is currently required for using CyteTypeR)
pbmc.markers <- FindAllMarkers(pbmc, only.pos = TRUE)

# Apply filtering criteria for markers
pbmc.markers %>%
  group_by(cluster) %>%
  dplyr::filter(avg_log2FC > 1)

```

## Running CyteTypeR
Like the original, CyteTypeR is run with 2 steps for efficiency:
* Step 1: prepare the data for job submission
* Step 2: Submit job and retrieve results
  
``` R
## Prep data for job submission to cytetype api
prepped_data <- PrepareCyteTypeR(pbmc,
         pbmc.markers,
         n_top_genes = 10,
         group_key = 'seurat_clusters',
         aggregate_metadata = TRUE,
         coordinates_key = "umap")

## Adding metadata on submission
metadata <- list(
  title = 'My scRNA-seq analysis of human pbmc',
  run_label = 'initial_analysis',
  experiment_name: 'pbmc_human_samples_study')


## Submit job to cytetype
pbmc.results <- CyteTypeR(obj=pbmc,
                          prepped_data = prepped_data, 
                          study_context = "pbmc blood samples from humans", 
                          metadata = metadata
                          )


```

## CyteType Results
A link to the CyteType report will be created for successfully submitted annotation jobs.

``` 
## Example output message
INFO [2025-09-03 13:49:51] Submitting job to https://nygen-labs-prod--cytetype-api.modal.run/annotate
INFO [2025-09-03 13:49:59] CyteType job (id: 6e081560-864c-492a-a1aa-138fa547f0fe) submitted. Polling for results...
INFO [2025-09-03 13:50:04] Report (updates automatically) available at:
https://nygen-labs-prod--cytetype-api.modal.run/report/6e081560-864c-492a-a1aa-138fa547f0fe
INFO [2025-09-03 13:50:04] If disconnected, retrieve results with: GetResults()
[DONE] [✔✔✔] 3/3 completed
INFO [2025-09-03 13:59:55] Job 6e081560-864c-492a-a1aa-138fa547f0fe completed successfully.

```
### Results table
CyteType results are saved under "misc" in the seurat object e.g.```seurat_obj@misc$cytetype_results```
Example of the results table: [cytetypeR_table_export.tsv](/inst/cytetypeR_table_export.tsv)
```
## View results table
> View(pbmc@misc[["cytetype_results"]])

## Cluster annotation and result are stored in each row, use names() to check result table fields.
> names(pbmc_small@misc[["cytetype_results"]])
 [1] "clusterId"            "annotation"           "ontologyTerm"         "granularAnnotation"   "cellState"           
 [6] "justification"        "supportingMarkers"    "conflictingMarkers"   "missingExpression"    "unexpectedExpression"

```




